### PR TITLE
Replace time-based names with UUID

### LINKED ISSUE #253: Do not use time-based names on group creation
We occasionally get tracebacks from automated tests on mozillians-stage because of conflicting group names when creating new groups. I guess that it's triggered because we are running tests in parallel and the group name is time based [1]. We should consider replacing that to something more random like `uuid4` [2].

[1] https://github.com/mozilla/mozillians-tests/blob/master/tests/test_group.py#L21
[2] https://docs.python.org/2/library/uuid.html

### PR DESCRIPTION
**Pull request by anu0012**
Fix #253

### CODE REVIEW

File: tests/test_group.py [hunk 1]
```diff
@@ -18,7 +19,7 @@ def test_group_description_edit(self, base_url, selenium, vouched_user):
         home_page.login(vouched_user['email'], vouched_user['password'])
 
         # Create a new group
-        group_name = (time.strftime('%x-%X'))
+        group_name = (uuid.uuid4())
```
Comment (davehunt) [inline tests/test_group.py]: The change looks good, but there are also three other uses of `time.strftime` in this file.

File: tests/test_group.py [hunk 2]
```diff
@@ -6,6 +6,7 @@
 
 import pytest
 import time
+import uuid
```
Comment (davehunt) [inline tests/test_group.py]: We should move this to the first block of imports so it complies with PEP8. Also, we can probably remove `import time` (which is also in the wrong place) once the other uses are replaced.

File: tests/test_group.py (hunk 1 above)
Comment (anu0012) [inline tests/test_group.py]: @davehunt I have fixes that. You can check whether it is ok or not.

File: tests/test_group.py [hunk 3]
```diff
@@ -4,8 +4,8 @@
 
 from random import randrange
 
+import uuid
```
Comment (davehunt) [inline tests/test_group.py]: Nit: Bump this up one more line so there's a blank line between system and third-party imports.

File: tests/test_group.py (hunk 1 above)
Comment (davehunt) [inline tests/test_group.py]: This is failing for me with `TypeError: object of type 'UUID' has no len()` is it passing for you?

Full trace:
```
self = <tests.test_group.TestGroup instance at 0x7fdca377aea8>
base_url = 'https://mozillians-dev.allizom.org'
selenium = <selenium.webdriver.remote.webdriver.WebDriver (session="614b2867a8c64a3485849e50b6629aed")>
vouched_user = {'email': '77ae49b1-364e-401b-9c38-8f38a02a6755@restmail.net', 'name': '77ae49b1', 'password': '1f72a184-bc07-4e52-a8f4-2f0da91c291c'}

    @pytest.mark.credentials
    def test_group_type_change(self, base_url, selenium, vouched_user):
        home_page = Home(base_url, selenium)
        home_page.login(vouched_user['email'], vouched_user['password'])
   
        # Create a new group
        group_name = (uuid.uuid4())
        settings = home_page.header.click_settings_menu_item()
>       group = settings.create_group(group_name)

tests/test_group.py:68:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
pages/settings.py:67: in create_group
    create_group.create_group_name(group_name)
pages/create_group_page.py:20: in create_group_name
    element.send_keys(group_name)
.tox/tests/lib/python2.7/site-packages/selenium/webdriver/remote/webelement.py:343: in send_keys
    local_file = self.parent.file_detector.is_local_file(*value)
.tox/tests/lib/python2.7/site-packages/selenium/webdriver/remote/file_detector.py:48: in is_local_file
    file_path = ''.join(keys_to_typing(keys))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

value = (UUID('3073eb64-3b51-4f44-b806-c1934d3c00b1'),)

    def keys_to_typing(value):
        """Processes the values that will be typed in the element."""
        typing = []
        for val in value:
            if isinstance(val, Keys):
                typing.append(val)
            elif isinstance(val, int):
                val = str(val)
                for i in range(len(val)):
                    typing.append(val[i])
            else:
>               for i in range(len(val)):
E               TypeError: object of type 'UUID' has no len()

.tox/tests/lib/python2.7/site-packages/selenium/webdriver/common/utils.py:150: TypeError
```

**Comment by stephendonner** [comment]
Hi @anu0012 are you still interested in/intending to work on this?

**Comment by anu0012** [comment]
@stephendonner Sure. Tell me what can be done to fix this.

**Comment by stephendonner** [comment]
@anu0012 great, thanks for responding so promptly!  @davehunt still has a couple pending questions for you, and this also is in need of a rebasing against/with master, thanks!

**Comment by davehunt** [comment]
I'm going to close this in favour of #290 due to lack of activity.

**Pull request closed without merging**